data outscale_net_peering crashes

---
name: Bug report
about: Let us know about an unexpected error, a crash, or an incorrect behavior.
labels: bug, new

---

<!--
Hi there,

Thank you for opening an issue. Please note that we try to keep the this tracker reserved for bug reports and feature requests.
To fix problems, we need clear reproduction cases - we need to be able to see it happen locally. A reproduction case is ideally something we can git-clone or copy-paste and run immediately, without inventing any details or context. 

* A short example can be directly copy-pasteable; longer examples should be in separate git repositories, especially if multiple files are needed
* Please include all needed context. For example, if you figured out that an expression can cause a crash, put the expression in a variable definition or a resource
* Set defaults on (or omit) any variables. The person reproducing it should not need to invent variable settings
* If multiple steps are required, such as running terraform twice, consider scripting it in a simple shell script. Providing a script can be easier than explaining what changes to make to the config between runs.
* Omit any unneeded complexity: remove variables, conditional statements, functions, modules, providers, and resources that are not needed to trigger the bug.
-->

### OpenTofu Version
<!---
Run `terraform version` to show the version, and paste the result between the ``` marks below.

If you are not running the latest version of Terraform, please try upgrading because your issue may have already been fixed.
-->

```
OpenTofu v1.9.1
on linux_amd64
+ provider registry.opentofu.org/outscale/outscale v1.1.3

```

### OpenTofu Configuration Files
<!--
Paste the relevant parts of your Terraform configuration between the ``` marks below.

For Terraform configs larger than a few resources, or that involve multiple files, please make a GitHub repository that we can clone, rather than copy-pasting multiple files in here.
-->

```terraform
data "outscale_net_peering" "monitoring" {
  filter {
    name = "source_net_account_ids"
    values = [var.source_net_peering]
  }
  filter {
    name = "state_names"
    values = ["pending-acceptance"]
  }
}
resource "outscale_net_peering_acceptation" "monitoring" {
    net_peering_id = data.outscale_net_peering.monitoring.net_peering_id
}

```

### Crash Output
<!--
If the console output indicates that Terraform crashed, please share a link to a GitHub Gist containing the output of the `crash.log` file.
-->
```
$ tofu plan

[...]

╷
│ Error: Request cancelled
│ 
│   with module.subnet_c.outscale_public_ip.nat_IP,
│   on modules/subnet/01-public-subnet.tf line 41, in resource "outscale_public_ip" "nat_IP":
│   41: resource "outscale_public_ip" "nat_IP" {
│ 
│ The plugin.(*GRPCProvider).ReadResource request was cancelled.
╵

Stack trace from the terraform-provider-outscale_v1.1.3 plugin:

panic: runtime error: index out of range [0] with length 0

goroutine 41 [running]:
github.com/outscale/terraform-provider-outscale/outscale.DataSourceOutscaleLinPeeringConnectionRead(0xc0000a8400, {0x11ebf60?, 0xc0007b8658?})
	github.com/outscale/terraform-provider-outscale/outscale/data_source_outscale_net_peering.go:118 +0x1405
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).read(0x17386f8?, {0x17386f8?, 0xc0008684e0?}, 0xd?, {0x11ebf60?, 0xc0007b8658?})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.36.1/helper/schema/resource.go:843 +0x15f
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).ReadDataApply(0xc0004fc800, {0x17386f8, 0xc0008684e0}, 0xc0000a8380, {0x11ebf60, 0xc0007b8658})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.36.1/helper/schema/resource.go:1075 +0x13a
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ReadDataSource(0xc000175170, {0x17386f8?, 0xc000868420?}, 0xc0008680f0)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.36.1/helper/schema/grpc_provider.go:1474 +0x6ac
github.com/hashicorp/terraform-plugin-mux/tf5muxserver.(*muxServer).ReadDataSource(0xc0000a9100, {0x17386f8?, 0xc000868150?}, 0xc0008680f0)
	github.com/hashicorp/terraform-plugin-mux@v0.18.0/tf5muxserver/mux_server_ReadDataSource.go:36 +0x18d
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ReadDataSource(0xc0000b8140, {0x17386f8?, 0xc000b89d70?}, 0xc0006a0fa0)
	github.com/hashicorp/terraform-plugin-go@v0.26.0/tfprotov5/tf5server/server.go:688 +0x26a
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ReadDataSource_Handler({0x1425c00, 0xc0000b8140}, {0x17386f8, 0xc000b89d70}, 0xc0006b0400, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.26.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:665 +0x1a6
google.golang.org/grpc.(*Server).processUnaryRPC(0xc00054e000, {0x17386f8, 0xc000b89d10}, 0xc00013aa20, 0xc000493650, 0x217fc80, 0x0)
	google.golang.org/grpc@v1.69.4/server.go:1392 +0xfbe
google.golang.org/grpc.(*Server).handleStream(0xc00054e000, {0x1739208, 0xc0005ce4e0}, 0xc00013aa20)
	google.golang.org/grpc@v1.69.4/server.go:1802 +0xb88
google.golang.org/grpc.(*Server).serveStreams.func2.1()
	google.golang.org/grpc@v1.69.4/server.go:1030 +0x7f
created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 51
	google.golang.org/grpc@v1.69.4/server.go:1041 +0x11d

Error: The terraform-provider-outscale_v1.1.3 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```


### Expected Behavior
I should fail with an error telling me no net peering correspond to the filters

### Actual Behavior
<!--
What actually happened?
-->
panic: runtime error:  (see crash log)

### Steps to Reproduce
<!--
Please list the full steps required to reproduce the issue, for example:
1. `terraform init`
2. `terraform apply`
-->
From another account, I created a net peering connection, accepted it from this account manually (cockpit), then deleted it. Recreated from other account.

Applied the tf configuration files, except I had only the first filter. Got an error because there was 2 net peering from the other account (deleted and new one), so I added the second filter. "tofu apply" worked, but then a new "tofu plan" got the "panic: runtime error".

Error goes away when I change the second filter to `["pending-acceptance", "active"]`  

### Additional Context
<!--
Are there anything atypical about your situation that we should know?
-->
region is cloudgouv-eu-west-1

--- a/outscale/data_source_outscale_net_peering.go
+++ b/outscale/data_source_outscale_net_peering.go
@@ -4,7 +4,6 @@ import (
 	"context"
 	"fmt"
 	"log"
-	"net/http"
 	"reflect"
 	"time"
 
@@ -93,7 +92,6 @@ func DataSourceOutscaleLinPeeringConnectionRead(d *schema.ResourceData, meta int
 	req.SetFilters(filtersValues)
 
 	var resp oscgo.ReadNetPeeringsResponse
-	var statusCode int
 	err = resource.Retry(5*time.Minute, func() *resource.RetryError {
 		rp, httpResp, err := conn.NetPeeringApi.ReadNetPeerings(context.Background()).ReadNetPeeringsRequest(req).Execute()
 
@@ -101,17 +99,15 @@ func DataSourceOutscaleLinPeeringConnectionRead(d *schema.ResourceData, meta int
 			return utils.CheckThrottling(httpResp, err)
 		}
 		resp = rp
-		statusCode = httpResp.StatusCode
 		return nil
 	})
-
 	if err != nil {
-		if statusCode == http.StatusNotFound {
-			return fmt.Errorf("no matching Net Peering Connection found")
-		}
 		return fmt.Errorf("Error reading Net Peering Connection details: %s", err)
 	}
 
+	if len(resp.GetNetPeerings()) == 0 {
+		return fmt.Errorf("No matching Net Peering Connection found")
+	}
 	if len(resp.GetNetPeerings()) > 1 {
 		return fmt.Errorf("multiple Net Peering connections matched; use additional constraints to reduce matches to a single Net Peering Connection")
 	}

--- a/tests/qa_provider_oapi/data/flexible_gpu_catalog/TF-176_flexible_gpu_catalog_datasource_attributes_ok/step1.flexible_gpu_catalog_datasource_attributes_ok.ref
+++ b/tests/qa_provider_oapi/data/flexible_gpu_catalog/TF-176_flexible_gpu_catalog_datasource_attributes_ok/step1.flexible_gpu_catalog_datasource_attributes_ok.ref
@@ -18,16 +18,6 @@
                         "flexible_gpu_catalog": [
                             {
                                 "generations": [
-                                    "v5",
-                                    "v6"
-                                ],
-                                "max_cpu": 35,
-                                "max_ram": 250,
-                                "model_name": "nvidia-a100",
-                                "v_ram": 40000
-                            },
-                            {
-                                "generations": [
                                     "v3",
                                     "v4"
                                 ],
@@ -38,11 +28,30 @@
                             },
                             {
                                 "generations": [
+                                    "v7",
+                                    "v104"
+                                ],
+                                "max_cpu": 130,
+                                "max_ram": 2300,
+                                "model_name": "nvidia-h200",
+                                "v_ram": 141000
+                            },
+                            {
+                                "generations": [
+                                    "v7"
+                                ],
+                                "max_cpu": 35,
+                                "max_ram": 256,
+                                "model_name": "nvidia-l40",
+                                "v_ram": 48000
+                            },
+                            {
+                                "generations": [
                                     "v5"
                                 ],
-                                "max_cpu": 80,
-                                "max_ram": 512,
-                                "model_name": "nvidia-p6",
+                                "max_cpu": 35,
+                                "max_ram": 250,
+                                "model_name": "nvidia-v100",
                                 "v_ram": 16000
                             },
                             {
@@ -57,22 +66,23 @@
                             },
                             {
                                 "generations": [
+                                    "v6",
                                     "v7"
                                 ],
                                 "max_cpu": 35,
                                 "max_ram": 256,
-                                "model_name": "nvidia-l40",
-                                "v_ram": 48000
+                                "model_name": "nvidia-a100-80",
+                                "v_ram": 80000
                             },
                             {
                                 "generations": [
-                                    "v7",
-                                    "v104"
+                                    "v5",
+                                    "v6"
                                 ],
-                                "max_cpu": 130,
-                                "max_ram": 2300,
-                                "model_name": "nvidia-h200",
-                                "v_ram": 0
+                                "max_cpu": 35,
+                                "max_ram": 250,
+                                "model_name": "nvidia-a100",
+                                "v_ram": 40000
                             },
                             {
                                 "generations": [
@@ -80,7 +90,7 @@
                                 ],
                                 "max_cpu": 80,
                                 "max_ram": 512,
-                                "model_name": "nvidia-p100",
+                                "model_name": "nvidia-p6",
                                 "v_ram": 16000
                             },
                             {
@@ -94,15 +104,6 @@
                             },
                             {
                                 "generations": [
-                                    "v5"
-                                ],
-                                "max_cpu": 35,
-                                "max_ram": 250,
-                                "model_name": "nvidia-v100",
-                                "v_ram": 16000
-                            },
-                            {
-                                "generations": [
                                     "v5",
                                     "v6"
                                 ],
@@ -113,13 +114,12 @@
                             },
                             {
                                 "generations": [
-                                    "v6",
-                                    "v7"
+                                    "v5"
                                 ],
-                                "max_cpu": 35,
-                                "max_ram": 256,
-                                "model_name": "nvidia-a100-80",
-                                "v_ram": 80000
+                                "max_cpu": 80,
+                                "max_ram": 512,
+                                "model_name": "nvidia-p100",
+                                "v_ram": 16000
                             }
                         ],
                         "id": "##id-0##",

